### PR TITLE
Fix CI by using a new `debian_minimal` rootfs image and updating some tests to work on Julia nightly

### LINKED ISSUE #46: GitHub Actions CI is failing on the `main` branch
See e.g. the test failures in:
- #45

### PR DESCRIPTION
**Pull request by DilumAluthge**
Fixes #46.

This PR fixes CI on both Julia 1.6 and Julia nightly.

### CODE REVIEW

**Comment by DilumAluthge** [comment]
@staticfloat So, compared to master (on which both 1.6 and nightly are failing), this PR does fix 1.6.

**Comment by DilumAluthge** [comment]
Okay, I'm getting closer to fixing CI on nightly.

**Comment by DilumAluthge** [comment]
@staticfloat This is now ready for review.

**Comment by staticfloat** [comment]
Darn, this IOBuffer change is quite annoying for us.  I'm tempted to change the Sandbox API to fix this.

**Comment by DilumAluthge** [comment]
> Darn, this IOBuffer change is quite annoying for us. I'm tempted to change the Sandbox API to fix this.

It is so annoying. And the changes in this PR make the test suite much more unreadable.

I like the idea of changing the Sandbox API to fix it. What would that entail?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ jobs:
       matrix:
         julia-version:
           - "1.6"
+          - "1"
```
Comment (giordano) [inline .github/workflows/ci.yml:19]: Why this?  Isn't this running twice on Julia v1.6 now?

Comment (DilumAluthge) [reply .github/workflows/ci.yml:19]: I was just putting this in for now with the assumption that 1.7 will be out relatively soon, and once that happens we'll want to test on both LTS and v1.

**Pull request merged**